src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -1355,45 +1163,45 @@ private Name emitTableSwitch(int pos, int numCases) {
         MethodType caseType = args.function.resolvedHandle().type()
             .dropParameterTypes(0, 1) // drop collector
             .changeReturnType(returnType);
-        String caseDescriptor = caseType.basicType().toMethodDescriptorString();
+        MethodTypeDesc caseDescriptor = methodDesc(caseType.basicType());
 
-        emitPushArgument(invoker, 2); // push cases
-        mv.visitFieldInsn(Opcodes.GETFIELD, "java/lang/invoke/MethodHandleImpl$CasesHolder", "cases",
-            "[Ljava/lang/invoke/MethodHandle;");
+        emitPushArgument(cob, invoker, 2); // push cases
+        cob.getfield(ClassDesc.ofInternalName("java/lang/invoke/MethodHandleImpl$CasesHolder"), "cases",
+                CD_MethodHandle.arrayType());
         int casesLocal = extendLocalsMap(new Class<?>[] { MethodHandle[].class });
```

Comment:
Yes, however actual API does not match directly.
It would require an extension of the API or more significant refactoring of InvokerBytecodeGenerator.